web/app.js

```diff
@@ -1333,6 +1338,57 @@ const PDFViewerApplication = {
 
     this._initializePageLabels(pdfDocument);
     this._initializeMetadata(pdfDocument);
+    this._initializeJavaScript(pdfDocument);
+  },
+
+  /**
+   * @private
+   */
+  async _initializeJavaScript(pdfDocument) {
+    if (!AppOptions.get("enableScripting")) {
+      return;
+    }
+    const objects = await pdfDocument.getFieldObjects();
+    const scripting = this.externalServices.scripting;
+
+    window.addEventListener("updateFromSandbox", function (event) {
+      const id = event.detail.id;
+      switch (id) {
```

Comment:
So the ID can be an actual element ID or these special values? That's a bit confusing; I'd expect ID to be undefined then and instead have a second `command` field for these commands.